multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1151,6 +1152,19 @@ def _fit(
         else:  # continuing training
             model = self._model
 
+        if OmegaConf.select(config, "env.compile.turn_on", default=False):
+            assert version.parse(torch.__version__) >= version.parse(TORCH_COMPILE_MIN_VERSION), (
+                f"torch.compile requires torch version >= {TORCH_COMPILE_MIN_VERSION}, "
+                f"but torch version {torch.__version__} is detected."
+            )
+            logger.debug("Using torch.compile() in compiling the model.")
+            model = torch.compile(
+                model,
+                mode=OmegaConf.select(config, "env.compile.mode", default="default"),
+                dynamic=OmegaConf.select(config, "env.compile.dynamic", default=True),
+                backend=OmegaConf.select(config, "env.compile.backend", default="inductor"),
```

Comment:
Do we want to use the default values in `default.yaml` instead? It can help in case we want to update the default values in future.

Reply:
Yes, `OmegaConf.select()` prefers using the values in default.yaml. Adding the default values in the function call is for backward compatibility, i.e., users load previously saved predictor which doesn't have these keys. Without these default values, it would throw an exception. See the doc: https://omegaconf.readthedocs.io/en/2.3_branch/usage.html#omegaconf-select

Reply:
I see. That makes sense.